marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs

```diff
@@ -227,79 +230,79 @@ class Provenance {
-    let provTypes = ['ps:Flow','ps:EntityProperty','ps:Mapping','dhf:Mapping'];
+  _createMappingStepRecord(jobId, flowId, docURI, info) {
+    let provId = `${jobId + flowId + 'mapping' + docURI}`;
+    let provTypes = ['ps:Flow','ps:Entity','dhf:Entity','dhf:MappingStep','dhf:MappingStepEntity'];
     if (info && info.status)
-      provTypes.push('dhf:Step' + this.hubutils.capitalize(info.status));
+      provTypes.push('dhf:Doc' + this.hubutils.capitalize(info.status));
 
     let recordOpts = {
       provTypes,
       relations: {
         associatedWith: flowId,
-        generatedBy: jobId || undefined,
+        generatedBy: jobId,
         derivedFrom: info && info.derivedFrom,
         influencedBy: info && info.influencedBy,
       },
-      // if we have a docURI, then we're recording the step execution on a document
-      // if not, then we're recording the state of the step itself changing
-      attributes: docURI ? { location: docURI } : undefined
+      attributes: { location: docURI }
     };
 
-    return this._createRecord(provId, recordOpts, info.metadata)
+    return (this.config.autoCommit) ? 
+      this._createRecord(provId, recordOpts, info.metadata) : 
+      this.config.commitQueue.push([provId, recordOpts, info.metadata]);
   }
 
   /**
    * @desc Create a provenance record when a document is run through a mastering step
-   * @param {string} [jobId] - the ID of the job being executed (unique), this will generate 
+   * @param {string} jobId - the ID of the job being executed (unique), this will generate 
    * @param {string} flowId - the unique ID of the flow
-   * @param {string} stepType - the type of step within a flow ['ingest','mapping','mastering','custom']
    * @param {string} [docURI] - the URI of the document being modified by this step
    * @param {Object} info
    * @param {string} info.derivedFrom - the entity, file or document URI that this ingested document was derived from
-   * @param {string} info.influencedBy - the mapping or component [aka. custom code] the document modified by
+   * @param {string} info.influencedBy - the mastering step the document was modified by
    * @param {string} [info.metadata] - key/value pairs to document with the provenance record
    *   Master Source doc property to Entity Instance property
    *    provTypes: [ "ps:Flow", "ps:EntityProperty", "dhf:Mastering" ],
    *    relations:
    *      - generatedBy (job id)
    *      - derivedFrom (uri)
    *      - derivedFrom (array of uris)
-   *      - influencedBy (mapping)
+   *      - influencedBy (mastering)
    *      - influencedBy (component [aka. custom code])
    *      - associatedWith (flow id)
    *    attributes:
    *      - location (doc URI)
    */
-  _createMasteringStepRecord(jobId, flowId, stepType, docURI, info) {
-    let provId = this.hubutils.uuid();
-    let provTypes = ['ps:Flow','ps:EntityProperty','dhf:Mastering'];
+  _createMasteringStepRecord(jobId, flowId, docURI, info) {
```

Comment:
```suggestion
  _createMasterStepRecord(jobId, flowId, docURI, info) {
```
We are using the step type "master" instead of "mastering", so I believe this needs to change for call on line 357 to work properly.